### PR TITLE
adds legacy_puppet_hostname setting

### PR DESCRIPTION
**Pull request by kitchen**
Resolves: redmine #1745
References: http://projects.theforeman.org/issues/1745

### CODE REVIEW

**Comment by skottler** [comment]
[test]

**Comment by domcleal** [comment]
Thanks for the PR @kitchen.  Would you mind adding a couple of tests to test/unit/smart_proxy_test.rb?

You'll find some similar ones in there that already check the URL, but you'd just need to test `proxy.to_s` instead.  You can call `Setting[:legacy_puppet_hostname] = true/false` before the test to change your setting on the fly, just add it to `test/fixtures/settings.yml` first (because the app/models/settings/ don't get loaded during tests).

**Comment by kitchen** [comment]
certainly! I'll probably have a bit of time tomorrow at the airport, I'll work on it. See you at puppetconf?

**Comment by domcleal** [comment]
Great, cheers.  Unfortunately not, but @skottler will be around.

**Comment by paradoxbound** [comment]
This worked for me building hosts on ec2 before I was just getting "server = puppet" now I am getting the fqdn of my puppet server.

**Comment by domcleal** [comment]
Oops, I didn't realise this had been updated.  Sorry for the delay.  Let's [test].

File: test/fixtures/settings.yml [hunk 1]
```diff
@@ -190,3 +190,8 @@ attribute38:
   category: Setting::Provisioning
   default: "true"
   description: "Should we use the originating IP of the built request to update the Host's IP?"
+attribute39:
+  name: legacy_puppet_hostname
+  category: Setting::Puppet
+  default: false
```
Comment (domcleal) [inline test/fixtures/settings.yml]: Can you try quoting this?  It's causing a test failure on MySQL which is usually due to data types.

Comment (kitchen) [inline test/fixtures/settings.yml]: sure, sorry about that!

**Comment by kitchen** [comment]
fixed the test fixture

**Comment by domcleal** [comment]
Cheers! [test]

**Comment by domcleal** [comment]
Thanks for the patch @kitchen!  Merged as f5d4e70 for Foreman 1.3.

**Pull request closed without merging**